src/native/corehost/fxr/fx_resolver.cpp

```diff
@@ -285,6 +288,58 @@ namespace
 
         return new fx_definition_t(fx_ref.get_fx_name(), selected_fx_dir, oldest_requested_version, selected_fx_version);
     }
+
+#ifdef TARGET_OSX
+    bool isRunningOnArm64()
+    {
+        int value = 0;
+        size_t len = sizeof(value);
+        int result = sysctlbyname("hw.optional.arm64", &value, &len, nullptr, 0);
+
+        return (result == 0) && (value != 0);
+    }
+#endif // TARGET_OSX
+
+    StatusCode check_framework_references_arch_compatible(
+        const fx_name_to_fx_reference_map_t& references)
+    {
+        StatusCode rc = StatusCode::Success;
+
+#ifdef TARGET_OSX
+        auto runtime_ref = references.find("Microsoft.NETCore.App");
+        assert(runtime_ref != references.end());
```

Comment:
The references are ordered, so there's no need to search for the root framework by name. We also don't want to hardcode the root framework name too much (testing purposes mostly). There's a helper `get_root_framework` which does what you want here (and is used for the same purpose in other places as well).